File: app/src/debug/java/com/orangepenguin/boilerplate/singletons/BoilerplateApplication.java
package com.orangepenguin.boilerplate.singletons;

import com.orangepenguin.boilerplate.Application;

import timber.log.Timber;

public class BoilerplateApplication extends Application {
    @Override
    protected void setUpLogging() {
        Timber.plant(new Timber.DebugTree() {
            @Override
            protected String createStackElementTag(StackTraceElement element) {
                return super.createStackElementTag(element) + ":" + element.getLineNumber();
            }
        });
    }
}

File: app/src/debug/java/com/orangepenguin/boilerplate/singletons/BoilerplateApplication.kt
package com.orangepenguin.boilerplate.singletons

import com.orangepenguin.boilerplate.Application

import timber.log.Timber

class BoilerplateApplication : Application() {
    override fun setUpLogging() {
        Timber.plant(object : Timber.DebugTree() {
            override fun createStackElementTag(element: StackTraceElement): String {
                return super.createStackElementTag(element) + ":" + element.lineNumber
            }
        })
    }
}
